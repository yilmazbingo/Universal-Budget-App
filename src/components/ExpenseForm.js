import React from "react";
import Form from "../components/formComponents/Form";
import SingleDatePicker from "./formComponents/singleDatePicker";
import Joi from "@hapi/joi";

class ExpenseForm extends Form {
  schema = Joi.object({
    description: Joi.string()
      .min(10)
      .alphanum()
      .label("Description")
      .required(),
    amount: Joi.number()
      .precision(2)
      .required()
      .label("Amount"),
    note: Joi.string()
      .min(5)
      .max(100)
      .label("Note")
  });
  onSubmit = this.props.onSubmit;
  render() {
    // console.log(this.props);
    console.log("schema changed", this.schema);
    return (
      <div>
        <form action="" onSubmit={this.handleSubmit}>
          {this.renderInput("description", "Description")}
          {this.renderInput("amount", "Amount", "number")}
          {this.renderTextarea("note", "Note")}
          {this.renderButton("Add Expense")}
          <SingleDatePicker />
        </form>
      </div>
    );
  }
}

export default ExpenseForm;
