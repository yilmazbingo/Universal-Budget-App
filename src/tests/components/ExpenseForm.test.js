import React from "react";
import { shallow } from "enzyme";
import ExpenseForm from "../../components/ExpenseForm";

test("should render ExpenseForm correctly", () => {
  const wrapper = shallow(<ExpenseForm />);
  expect(wrapper).toMatchSnapshot();
});

//window.alert should be mocked
// test("should render error for invalied form submission", () => {
//   const wrapper = shallow(<ExpenseForm />);
//   wrapper.find("form").simulate("submit", { preventDefault: () => {} });
// });
