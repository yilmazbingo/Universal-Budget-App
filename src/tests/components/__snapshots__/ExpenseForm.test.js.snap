// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseForm correctly 1`] = `
<div>
  <form
    action=""
    onSubmit={[Function]}
  >
    <Input
      autoFocus={true}
      label="Description"
      name="description"
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <Input
      autoFocus={true}
      label="Amount"
      name="amount"
      onChange={[Function]}
      placeholder=""
      type="number"
      value=""
    />
    <Textarea
      label="Note"
      name="note"
      onChange={[Function]}
      placeholder=""
      value=""
    />
    <button>
      Add Expense
    </button>
    <DatePicker />
  </form>
</div>
`;
